judge/views/problem_data.py

```diff
@@ -166,15 +178,15 @@ def get_valid_files(self, data, post=False):
             elif data.zipfile:
                 return ZipFile(data.zipfile.path).namelist()
         except BadZipfile:
-            return []
+            return None
         return []
 
     def get_context_data(self, **kwargs):
         context = super(ProblemDataView, self).get_context_data(**kwargs)
         if 'data_form' not in context:
             context['data_form'] = self.get_data_form()
             valid_files = context['valid_files'] = self.get_valid_files(context['data_form'].instance)
-            context['data_form'].zip_valid = valid_files is not False
```

Comment:
When could the old code return `False` for this? Is this just... completely borked?

Reply:
Yeah, it sort of looks like it currently always returns "yep, valid zip".

Reply:
Ok, updated and tested. Now we correctly display errors for invalid zip files, instead of prod which does:
![image](https://github.com/DMOJ/online-judge/assets/41458184/af2f6910-9378-4415-b218-0da4f7b699e5)